File: school_Management_Backend/src/main/java/com/school/management/service/academic/HourSlotService.java
package com.school.management.service.academic;

import com.school.management.dto.academic.HourSlotCreateDTO;
import com.school.management.dto.academic.HourSlotResponseDTO;
import java.util.List;

public interface HourSlotService {
    HourSlotResponseDTO addHourSlot(HourSlotCreateDTO dto);
    List<HourSlotResponseDTO> getSchoolHourSlots(Long schoolId);
    HourSlotResponseDTO updateHourSlot(Long schoolId, Long id, HourSlotCreateDTO dto);
    void deleteHourSlot(Long schoolId, Long id);
}

File: school-management-frontend/src/services/pedagogieService/hourSlotService.js
import api from '../api';

const API_URL = '/v1/hour-slots';

const hourSlotService = {
  // Créer une nouvelle tranche horaire
  addHourSlot: async (slotData) => {
    const response = await api.post(API_URL, slotData);
    return response.data;
  },

  // Récupérer les tranches horaires d'une école
  getSchoolHourSlots: async (schoolId) => {
    const response = await api.get(API_URL, {
      headers: { 'X-School-Id': schoolId }
    });
    return response.data;
  },

  // Modifier une tranche horaire
  updateHourSlot: async (schoolId, id, slotData) => {
    const response = await api.put(`${API_URL}/${id}`, slotData, {
      headers: { 'X-School-Id': schoolId }
    });
    return response.data;
  },

  // Supprimer une tranche horaire
  deleteHourSlot: async (schoolId, id) => {
    const response = await api.delete(`${API_URL}/${id}`, {
      headers: { 'X-School-Id': schoolId }
    });
    return response.data;
  }
};

export default hourSlotService;